### PR TITLE
#8 - Do not send BAM configuration unless the poller is link to 1 BA …

### PR DESCRIPTION
**Pull request by Moujimouja**
Fix poller display when BAM is not installed on the poller display but installed on the Central

### CODE REVIEW

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 1]
```diff
@@ -102,7 +102,25 @@ public function isBamModuleAvailable()
      */
     public function generateObjects($poller_id)
     {
-        $sql = '';
+
+      #Check that the poller id is used as a BAM poller display to prevent
+      #a BAM configuration to be sent to a  poller display without Bam
+      $isBamPoller = false;
+
+      $queryIsPollerUsedWithBAM = "SELECT count(*) as nbBa FROM mod_bam_poller_relations where poller_id = ".$poller_id;
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: space around dot

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 2]
```diff
@@ -102,7 +102,25 @@ public function isBamModuleAvailable()
...
+      $IsPollerUsedWithBAM = $this->backend_instance->db->query($queryIsPollerUsedWithBAM);
+
+      $rowIsPollerUsedWithBAM = $IsPollerUsedWithBAM->fetch();
+
+      if ((($rowIsPollerUsedWithBAM !== false) && $rowIsPollerUsedWithBAM['nbBa'] > 0)) {
+        $isBamPoller = true;
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: 4 spaces indentation

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 3]
```diff
@@ -102,7 +102,25 @@ public function isBamModuleAvailable()
...
+      }
+
+      #If the poller is used as an additionnal poller for BAM, he should have
+      #Centreon BAM installed so we can continue
+      $sql = '';
+
+      if($isBamPoller){
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: spaces around parenthesis

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 4]
```diff
@@ -187,7 +205,7 @@ public function generateObjects($poller_id)
 
         // Enable MySQL foreign key check
         $sql .= $this->setForeignKey(1). "\n\n";
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: 4 lines indentation

**Review by kduret — CHANGES_REQUESTED** [review]
Bad coding style. You need to respect PSR-2

**Comment by Moujimouja** [comment]
Is it better ?

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
- * Centreon also meet, for each linked independent module, the terms  and conditions
- * of the license of that module. An independent module is a module which is not
- * derived from this program. If you modify this program, you may extend this
- * exception to your version of the program, but you are not obliged to do so. If you
- * do not wish to do so, delete this exception statement from your version.
- *
- * For more information : contact@centreon.com
- *
- */
+* Copyright 2005-2017 Centreon
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: you have to indent of 1 space

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 6]
```diff
@@ -52,34 +52,34 @@
 use \CentreonPollerDisplayCentral\ConfigGenerate\Bam\Kpi;
 
 /**
- * User: kduret
- * Date: 23/02/2017
- * Time: 09:19
- */
+* User: kduret
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: you have to indent of 1 space

File: www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php [hunk 7]
```diff
@@ -52,34 +52,34 @@
...
+* Date: 23/02/2017
+* Time: 09:19
+*/
 class Bam extends \AbstractObject
 {
     /**
-     *
-     * @var boolean
-     */
+    *
```
Comment (kduret) [inline www/modules/centreon-poller-display-central/core/class/configGenerate/Bam.php]: you have to indent of 1 space

**Review by kduret — CHANGES_REQUESTED** [review]

**Pull request merged**